### DISCUSSION TITLE
Powering ESP-01?

### DISCUSSION THREAD

**Discussion by jimford-sudo** [arendst/Tasmota#17861] [Q&A]
After Tasmotizing some ESP-01 modules, I tried powering them from an external 3.3V source rather than from the device I used for Tasmotising - but they failed even though I was careful in setting my variable power supply to 3.3v.

What did I do wrong?
I would be grateful for some help, please.

Jim

**Comment by barbudor** [comment]
I gave up using ESP-01 for that
It was working correctly when connected to the flasher but once connected to a separated power supply, even with additional capacitors, it was totally random behavior
It was direct to the bin for me

**Reply by barbudor** [reply]
Some ESP modules are lacking the pull up on GPIO 0 and EN which allows proper boot
If GPIO0 (and EN) are connected to your flasher that may also explain why it works when connected to the flasher but not stand-alone.
Are you facing a random behavior or a not-start at all ?

**Reply by jimford-sudo** [reply]
Mine won't boot at all in stand-alone. 
What value resistor should I use to pull-up GPIO 0 and EN and will it preclude using GPIO 0 for other purposes?

Thanks for the replies
Jim

**Reply by barbudor** [reply]
You can use around 10-15k for gpio0 pull
If you have no need for EN, just tight it with a small wire to RST near by which already has a pull up
If you are using gpio2 and there is a risk that what is connect is pulling gpio2 down, better to add pull up to gpio2 too

I've tried again yesterday because I still have 2 in a box:
- working when powered by the USB flasher
- crashing when powered by another power supply (I kept TX and gnd connected to the USB serial adapter to see the log and it was crashing in loop)

So definitively very very sensitive to the power supply

**Reply by barbudor** [reply]
Exemple of adding
- Pullups (12K) on GPIO0 and GPIO2
- Solder blob between EN and RST

Powered from a lab power supply, looks stable (tasmota-lite, sleep 150)

![IMG_20230203_214843](https://user-images.githubusercontent.com/2996042/216710358-0f3acf36-af6d-42f9-b535-c4bfae220c1a.jpg)

**Reply by Jason2866** [reply]
Missing caps. One (> 10uF) to stabilise power and some with a few nF to block HF noise.
But the easy reachable GPIOs are a very bad design decision

**Reply by jimford-sudo** [reply]
In view of my and others experience of the ESP-01, I've abandoned the idea of using one and decided to go with w Wemos Mini D1.

Many thanks for the helpful replies. Jim

**Reply by Jason2866** [reply]
@jimford-sudo replaced worst with a bad one. Get a NodeMCU. Wemos Mini D1 clones have nearly all to weak LDO. [See](https://github.com/arendst/Tasmota/issues/17500#issuecomment-1416846464)

**Comment by jimford-sudo** [comment]
Thanks for the suggestion Jason - I've just ordered one.
Jim

**Comment by wchulze** [comment]
![image](https://user-images.githubusercontent.com/41591541/217079931-3fc7656d-442f-4c40-aa22-95354f61f2c7.png)

Schematic of ESP01S: C2, C3, C4 are ok

![image](https://user-images.githubusercontent.com/41591541/217080612-9c1d2193-0d41-433b-a726-b4b6f27f56ac.png)

Runnig this sinece a couple of years without problem. Not Tasmota related but Wify stack uesd. Hope it help.

![image](https://user-images.githubusercontent.com/41591541/217081518-cccae78c-e1e1-4239-b3a2-fb8de4c64cbc.png)
